src/libvirtApi/domain.js

```diff
@@ -613,17 +604,11 @@ export function domainInstall({ onAddErrorNotification, vm }) {
     const args = JSON.stringify({
         autostart: vm.autostart,
         connectionName: vm.connectionName,
-        disks: prepareDisksParam(vm.disks),
-        firmware: vm.firmware == "efi" ? 'uefi' : '',
```

Comment:
So we should first introduce a test that switches back from uefi to bios, make sure that works on main, and that should then fail here, right?

Reply:
yeah, in our test we only tests we currently only switch from BIOS -> UEFI. But This bug is mostly about switching from BIOS -> BIOS, and I think that way maybe incorrect value gets written into XML for BIOS (didn't investigate further, not sure by that claim).
I'm not sure if a test for BIOS -> UEFI -> BIOS would bring anything meaningful, but it would not be hard to code.

Reply:
I sent PR #656 for this -- that detects this regression.

Reply:
Merely adding back the property setting here does not fix this. So this needs some more work. I included the test case here, so that we have a reminder.

Reply:
I fixed the state vs. xml values more properly now. Works fine locally/interactively, I'll let the tests confirm.